core/arch/arm/mm/tee_mmu.c

```diff
@@ -757,8 +761,15 @@ void teecore_init_pub_ram(void)
 	tee_mm_final(&tee_mm_pub_ddr);
 	tee_mm_init(&tee_mm_pub_ddr, s, s + nsec_tee_size, SMALL_PAGE_SHIFT,
 		    TEE_MM_POOL_NO_FLAGS);
-
 	s += nsec_tee_size;
+
+#ifdef CFG_PL310
+	/* Allocate statically the l2cc mutex */
+	TEE_ASSERT((e - s) > 0);
+	tee_l2cc_store_mutex_boot_pa(s);
+	s += sizeof(uint32_t);		/* size of a pl310 mutex */
```

Comment:
Shouldn't matter, we're mapping this using 1 meg sections any way and the normal world shared memory allocator should round up to some usable address.